draft-ietf-ppm-dap.md

```diff
@@ -2214,7 +2202,7 @@ from its query and the response to its query as follows:
 * For time_interval tasks, the batch selector is the batch interval specified in
   the query.
 
-* For fixed_size tasks, the batch selector is the batch ID assigned sent in the
+* For leader_selected tasks, the batch selector is the batch ID sent in the
```

Comment:
The aggregate share AAD includes the batch ID, so the batch ID needs to be communicated to the Collector to allow decryption, alas. I restored the batch ID to the `Collection` message.

However, I think maybe this isn't the right fix (just the least controversial one to unbreak things for now). DAP aims to provide robustness only if both aggregators are honest (and robustness is trivally breakable by a single dishonest aggregator -- either aggregator can forge results). If both aggregators are honest, there is no need for the aggregate share AAD to include the batch ID, since there is no need for the Collector to e.g. detect the aggregator attempting to repeatedly transmit an aggregate share. So I think the "right" fix would be to drop batch ID from both the `Collection` and `AggregateShareAad` messages.

However, this argument applies just as well to every other field in `AggregateShareAad` -- so what is the purpose of the AAD here? (best-effort protection against bugs, perhaps?)

If this isn't trivially resolvable, this might be worthy of an issue to allow discussion on what the right fix is.

Reply:
(I think the same argument doesn't apply to `InputShareAad`, since this AAD is used to allow the Helper to ensure the encrypted report share it received from the Leader matches the rest of the report share.)

Reply:
I think we should include the batch ID in the AAD. It's not strictly necessary in our threat model, so think of it as defense in depth: successful execution implies agreement on the batch ID between the aggregators and collector; this may have benefits we don't see right now.

As a general rule, binding protocol execution to crypto operations is a good idea whenever feasible. There are tons of examples of this, the most recent of which might be https://eprint.iacr.org/2023/691.

Reply:
Fair enough. Even if this AAD doesn't support the explicit threat model, at the very least it will help avoid accidental bugs which e.g. mismatch. (I'm not sure it can protect against any form of malice, as the components of the AAD are public to both  aggregators; and each aggregator can already falsify results of any aggregate shares it can control -- i.e. the Helper can falsify the Helper's aggregate share, the Leader can falsify the Leader or Helper's aggregate share.)